src/map/skill.hpp

```diff
@@ -341,9 +341,19 @@ struct skill_timerskill {
 	int flag;
 };
 
-#define MAX_SKILLUNITGROUP 25 /// Maximum skill unit group (for same skill each source)
+/// Skill unit
+struct skill_unit {
+	struct block_list bl;
+	std::shared_ptr<s_skill_unit_group> group; /// Skill group reference
```

Comment:
We need to make sure this is properly initialized on allocation, since this is still outside of the "clean" C++ scope.

Reply:
Cast skill -> skill_unitsetting -> skill_initunitgroup ('group' is std::<make_shared>() here) -> skill_initunit ('group' is passed here to the skill_unit struct)